test/unit/plugins/rest_spec.js

```diff
@@ -68,25 +54,59 @@ describe("The Rest plugin", function () {
 	});
 
 	describe("creating a new instance", function () {
-		var mapper = new MemoryMapper();
-		var server = new Hapi.Server();
+		var server          = new Hapi.Server();
+		var instanceAdapter = new InstanceAdapter();
+		var createInstanceStub;
+		var runInstancesStub;
+		var updateInstanceStub;
+
+		var awsAdapter = new AwsAdapter();
 
 		before(function () {
+			createInstanceStub = Sinon.stub(instanceAdapter, "createInstance")
+			.returns(Bluebird.resolve({
+				ami   : "ami-d05e75b8",
+				type  : "t2.micro",
+				state : "pending",
+				uri   : ""
+			}));
+
+			updateInstanceStub = Sinon.stub(instanceAdapter, "updateInstance")
+			.returns(Bluebird.resolve({
+				ami   : "ami-d05e75b8",
+				type  : "t2.micro",
+				state : "failed",
```

Comment:
Hard-coding the mock to always returned the `failed` status seems to gloss over the fact that we need the REST API to explicitly set this state. We need our test cases to demonstrate that this is done correctly.